.github/workflows/utils/setup_self_hosted_macos_env/action.yml

```diff
@@ -0,0 +1,117 @@
+name: Setup PATH for Self-Hosted MacOS Runner
+description: |
+  This workflow sets up the workflow environment for the Self-Hosted MacOS runner. 
+  It should be called right after actions/checkout.
+
+
+inputs:
+  python_version:
+    description: Version of Python being used by the Job
+    required: true
+
+outputs:
+  python_version:
+    description: 'The version of Python that was setup'
+    value: ${{ steps.python_venv.outputs.python_version }}
+  python_path:
+    description: 'Full absolute path to the python binary'
+    value: ${{ steps.python_venv.outputs.python_path }}
+  venv_location:
+    description: 'The path to the virtualenv created for python'
+    value: ${{ steps.python_venv.outputs.venv_location }}
+  venv_site_packages:
+    description: 'The path to the site-packages folder for the python venv created'
+    value: ${{ steps.site_packages.outputs.site_packages_dir }}
+  dependency_build_dir:
+    description: 'The directory where the LLVM/MLIR/MHLO sources should reside in'
+    value: ${{ steps.build_dir.outputs.dir_name }}
+
+
+runs:
+  using: composite
+  steps:
+
+    # These environment variables ensure multi-threading workloads (ex: pytest-xdist)
+    # work without any issues. Without them, the tests take hours to complete.
+    - name: Setup MacOS Env Vars
+      shell: bash
+      run: |
+        echo 'no_proxy=*' >> $GITHUB_ENV
+        echo 'OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES' >> $GITHUB_ENV
+        echo 'LDFLAGS=-L/opt/homebrew/opt/libomp/lib' >> $GITHUB_ENV
+        echo 'CPPFLAGS=-I/opt/homebrew/opt/libomp/include' >> $GITHUB_ENV
```

Comment:
These flags might also deserve a comment.

Reply:
Sure thing, adding these as environment variables was actually in the output when you install libomp which is why I added these.

Reply:
I've added a comment around these flags